src/core/runtime.d

```diff
@@ -328,46 +354,58 @@ struct Runtime
      *
      *             if (fp)
      *             {
+     *                 ++nRun;
      *                 try
      *                 {
      *                     fp();
      *                 }
      *                 catch (Throwable e)
      *                 {
      *                     writeln(e);
-     *                     failed++;
+     *                     --failed;
      *                 }
      *             }
      *         }
      *     }
-     *     return failed == 0;
+     *     return failed == 0 ? nRun : failed;
      * }
      * ---------
      */
     static @property void moduleUnitTester( ModuleUnitTester h )
     {
         sm_moduleUnitTester = h;
+        sm_boolModuleUnitTester = null;
     }
 
+    /// Ditto
+    static @property void moduleUnitTester( BoolModuleUnitTester h )
+    {
+        sm_moduleUnitTester = null;
+        sm_boolModuleUnitTester = h;
+    }
 
     /**
-     * Gets the current module unit tester.
+     * Gets the current module unit tester. Note that if a boolean returning
+     * module unit tester is set, it will be returned, even though the type is
+     * not exactly the same.
      *
      * Returns:
      *  The current module unit tester handler or null if none has been set.
      */
     static @property ModuleUnitTester moduleUnitTester()
     {
+        if(sm_moduleUnitTester == null)
+            return cast(ModuleUnitTester)sm_boolModuleUnitTester;
```

Comment:
So I'm a bit concerned about doing it that way. I don't know what code exists out there, but there is the possibility that code is checking to see if their particular unit tester is set. If I wrap in a lambda, that will always be false.

Now, code that checks against the function address of wrong type is going to fail to compile. But there will be no way to rectify the situation. At least with the way I have implemented, you can cast to the correct type to figure out if you've set the function.

Alternatively, I can just expose the "new" style unit tester via a new property (and make them mutually exclusive).